src/application/index.tsx

```diff
@@ -73,9 +75,18 @@ const cache = new InMemoryCache({
   },
 });
 
+const rpc = getPanelRpc(window);
+
 const cacheVar = makeVar<string | null>(null);
 export const client = new ApolloClient({
   cache,
+  resolvers: {
+    Query: {
+      memoryInternals(): Promise<MemoryInternals | undefined> {
+        return rpc.request("getMemoryInternals");
+      },
+    },
+  },
```

Comment:
I don't know if this is genius or incredibly stupid, but it might be very interesting in combination with `pollInterval`.
@jerelmiller do you have an opinion on this?

Reply:
I think this is totally fine! And agreed. The more I'm working with devtools, the more I think we can move some of the code in `devtools.ts` over to the React application, This falls in line with my thinking there so I dig it.